lib/hex.ml

```diff
@@ -35,7 +45,16 @@ let to_char x y =
   in
   Char.chr (code x lsl 4 + code y)
 
-let of_helper ?(ignore=[]) (next : int -> char) len =
+let of_string_fast s =
+  let len = String.length s in
+  let buf = Bytes.create (len * 2) in
+  for i = 0 to len - 1 do
+    Bytes.set buf       (i * 2)  hexa1.[Char.code s.[i]];
```

Comment:
It's ok by me.  There are six bounds checked each time through the loop (two sets and four gets); turning them all off brings the execution time down quite a bit.